Adaptors/Memory/src/ResultTable.cs

```diff
@@ -308,4 +223,47 @@ public Task<HealthCheckResult> Check(HealthCheckTag tag)
     => Task.FromResult(isInitialized_
                          ? HealthCheckResult.Healthy()
                          : HealthCheckResult.Unhealthy());
+
+  /// <inheritdoc />
+  public Task<long> UpdateManyResults(Expression<Func<Result, bool>>                                              filter,
+                                      ICollection<(Expression<Func<Result, object?>> selector, object? newValue)> updates,
+                                      CancellationToken                                                           cancellationToken = default)
+  {
+    long i = 0;
+    foreach (var session in results_.Values)
+    {
+      foreach (var id in session.Values.AsQueryable()
+                                .Where(filter)
+                                .Select(data => data.ResultId))
+      {
+        i++;
+        session.AddOrUpdate(id,
+                            _ => throw new ResultNotFoundException("Result not found"),
+                            (_,
+                             data) => new Result(data,
+                                                 updates));
+      }
+    }
+
+    return Task.FromResult(i);
+  }
+
+  /// <inheritdoc />
+  public Task<Result> UpdateOneResult(string                                                                      sessionId,
+                                      string                                                                      resultId,
+                                      ICollection<(Expression<Func<Result, object?>> selector, object? newValue)> updates,
+                                      CancellationToken                                                           cancellationToken = default)
+  {
+    if (!results_.TryGetValue(sessionId,
+                              out var session))
+    {
+      throw new SessionNotFoundException($"Session '{sessionId}' not found");
+    }
+
+    return Task.FromResult(session.AddOrUpdate(resultId,
```

Comment:
It should be result before update. (It is what's explained in ITaskTable doc). Maybe it returns the value before update ?
If this is based on what's done in TaskTable implementation, there may be an issue and a missing test

Reply:
Yes, it is based on the TaskTable implementation. I will fix both.